src/engraving/rendering/dev/arpeggiolayout.cpp

```diff
@@ -55,3 +59,276 @@ void ArpeggioLayout::layoutOnEdit(Arpeggio* item, LayoutContext& ctx)
+    if (item->arpeggioType() == ArpeggioType::DOWN) {
+        bottom -= item->symBbox(SymId::wiggleArpeggiatoUpArrow).width();
+    } else if (item->arpeggioType() == ArpeggioType::DOWN_STRAIGHT) {
+        bottom -= item->symBbox(SymId::arrowheadBlackDown).width();
+    }
+
+    return bottom;
+}
+
+//---------------------------------------------------------
+//   insetWidth
+//---------------------------------------------------------
+
+double ArpeggioLayout::insetWidth(Arpeggio* item)
+{
+    switch (item->arpeggioType()) {
+    case ArpeggioType::NORMAL:
+    {
+        return 0.0;
+    }
+
+    case ArpeggioType::UP:
+    case ArpeggioType::DOWN:
+    {
+        // use wiggle height, not width, since it's rotated 90 degrees
+        return (item->width() - item->symBbox(SymId::wiggleArpeggiatoUp).height()) / 2;
+    }
+
+    case ArpeggioType::UP_STRAIGHT:
+    case ArpeggioType::DOWN_STRAIGHT:
+    {
+        return (item->width() - item->style().styleMM(Sid::ArpeggioLineWidth)) / 2;
+    }
+
+    case ArpeggioType::BRACKET:
+    {
+        return item->width() - item->style().styleMM(Sid::ArpeggioLineWidth) / 2;
+    }
+    }
+    return 0.0;
+}
+
+//---------------------------------------------------------
+//   clearAccidentals
+//
+//   This function finds collisions with accidentals outside of an arpeggio's span
+//   Assumptions are made to deal with cross staff height, as we don't know this yet.
+//   Once a chord is found, its shape is stripped down to only accidentals which collide
+//   with the arpeggio.  The arpeggio's offset is calculated from the remaining shape
+//---------------------------------------------------------
+
+void ArpeggioLayout::clearAccidentals(Arpeggio* item, LayoutContext& ctx)
+{
+    if (!item || !item->chord()) {
+        return;
+    }
+
+    const Segment* seg = item->chord()->segment();
+    const Chord* endChord = toChord(seg->elementAt(item->endTrack()));
+    if (!endChord) {
+        endChord = item->chord();
+    }
+    const Part* part = item->part();
+    const track_idx_t partStartTrack = part->startTrack();
+    const track_idx_t partEndTrack = part->endTrack();
+
+    double accidentalDistance = ctx.conf().styleMM(Sid::ArpeggioAccidentalDistance) * item->mag();
+    Shape arpShape = item->shape().translate(item->pagePos());
+    double arpX = arpShape.right();
+    double largestOverlap = 0.0;
+
+    for (track_idx_t track = partStartTrack; track < partEndTrack; ++track) {
+        EngravingItem* e = seg->element(track);
+        if (e && e->isChord()) {
+            Chord* chord = toChord(e);
+            // Only check chords the arpeggio doesn't span
+            // We already calculated the gap between chords it does span in ChordLayout::layoutPitched
+            bool aboveStart = chord->vStaffIdx() <= item->vStaffIdx() && chord->line() < item->chord()->line();
+            bool belowEnd = chord->vStaffIdx() >= endChord->vStaffIdx() && chord->line() > endChord->line();
```

Comment:
Don't you want to use `upLine` and `downLine` explicitly, instead of just `line` which decides based on the stem direction (which seems irrelevant to me here)?

Also, shouldn't the logic be like this:
```suggestion
            bool aboveStart = chord->vStaffIdx() < item->vStaffIdx() ||(chord->vStaffIdx() == item->vStaffIdx() && chord->line() < item->chord()->line());
            bool belowEnd = chord->vStaffIdx() > endChord->vStaffIdx() || (chord->vStaffIdx() == endChord->vStaffIdx() && chord->line() > endChord->line());
```
which might be written more concisely as
```suggestion
            bool aboveStart = std::make_pair(chord->vStaffIdx(), chord->line()) < std::make_pair(item->vStaffIdx(), item->chord()->line());
            bool belowEnd = std::make_pair(chord->vStaffIdx(), chord->line()) > std::make_pair(endChord->vStaffIdx(), endChord->line());